src/xbot/common/properties/DatabaseStorageBase.java

```diff
@@ -13,25 +13,140 @@
  *
  * @author John
  */
-public abstract class DatabaseStorageBase extends PermanentStorageProxy {
+public abstract class DatabaseStorageBase implements ITableProxy {
 
     private static Logger log = Logger.getLogger(DatabaseStorageBase.class);
 
     private String dbUrlPreFormat = "jdbc:derby:%1s;create=true";
     private final String dbUrl;
+    private Connection conn;
 
     public DatabaseStorageBase(String databaseDirectory) {
-        super();
-
+        
         dbUrl = String.format(dbUrlPreFormat, databaseDirectory);
+        
+        try {
+            conn = DriverManager.getConnection(dbUrl);
+        } catch (SQLException e) {
+            log.error("Ran into a SQL problem - could not open a connection to the database! No properties will be loaded or persisted!!");
+            log.error(e.toString());
+            e.printStackTrace();
+        } catch (Exception e) {
+            log.error("Ran into a general expection, so could not open a connection to the database! No properties will be loaded or persisted!!");
+            log.error(e.toString());
+            e.printStackTrace();
+        }        
+        
+        // create properties if not exists
+        if (propertiesTableExists() == false)
+        {
+            createPropertiesTable();
+        }
+    }
+    
+    public void setDouble(String key, double value){
+        saveProperty("double", key, Double.toString(value));
+    }
+    
+    public void setBoolean(String key, boolean value){
+        saveProperty("boolean", key, Boolean.toString(value));
+    }
+    
+    public void setString(String key, String value){
+        saveProperty("string", key, value);
+    }
+    
+    public Double getDouble(String key){
+        String value = loadProperty(key);
+        try {
+            return Double.parseDouble(value);
+        }
+        catch (NumberFormatException e)
+        {
+            return null;
+        }
+    }
+    
+    public Boolean getBoolean (String key){
+        String value = loadProperty(key);
+        return parseBoolean(value);
+    }
+    
+    private Boolean parseBoolean(String value)
+    {
+        if (value.toLowerCase().equals("true"))
+        {
+            return true;
+        }
+        
+        if (value.toLowerCase().equals("false"))
+        {
+            return false;
+        }
+        
+        return null;
+    }
+    
+    public String getString (String key){
+        String value = loadProperty(key);
+        if (value.length() == 0)
+        {
+            return null;
+        }
+        return value;
+    }
+    
+    public void clear(){
+        obliterateStorage();
+    }
+    
+    private void saveProperty(String type, String name, String value) {
+       
+        String payload = "UPDATE PROPERTIES SET TYPE='" + name + "', VALUE='" + type + "' WHERE NAME = '" + value + "'";
```

Comment:
A little sanitization here would make a lot of sense, if nothing else make sure no ' ?